forest/cli/src/cli/snapshot_cmd.rs

```diff
@@ -0,0 +1,329 @@
+                let chain_name = chain_get_name().await.map_err(handle_rpc_err).unwrap();
+
+                let vars = HashMap::from([
+                    ("year".to_string(), year.to_string()),
+                    ("month".to_string(), month_string),
+                    ("day".to_string(), day_string),
+                    ("chain".to_string(), chain_name),
+                    ("height".to_string(), epoch.to_string()),
+                ]);
+                let output_path = match strfmt(&output_path.display().to_string(), &vars) {
+                    Ok(path) => path.into(),
+                    Err(e) => {
+                        cli_error_and_die(format!("Unparsable string error: {}", e), 1);
+                    }
+                };
+
+                let params = (
+                    epoch,
+                    *recent_stateroots,
+                    *include_old_messages,
+                    output_path,
+                    TipsetKeysJson(chain_head.key().clone()),
+                    *skip_checksum,
+                );
+
+                // infallible unwrap
+                let out = chain_export(params).await.map_err(handle_rpc_err).unwrap();
+
+                println!("Export completed. Snapshot located at {}", out.display());
+            }
+            Self::Fetch { snapshot_dir } => {
+                let snapshot_dir = snapshot_dir
+                    .clone()
+                    .unwrap_or_else(|| default_snapshot_dir(&config));
+                match snapshot_fetch(&snapshot_dir, config).await {
+                    Ok(out) => println!("Snapshot successfully downloaded at {}", out.display()),
+                    Err(e) => cli_error_and_die(format!("Failed fetching the snapshot: {e}"), 1),
+                }
+            }
+            Self::Dir => {
+                let dir = default_snapshot_dir(&config);
+                println!("Default snapshot dir: {:?}", dir);
+            }
+            Self::List { snapshot_dir } => {
+                list(&config, snapshot_dir);
+            }
+            Self::Remove {
+                filename,
+                snapshot_dir,
+                force,
+            } => {
+                remove(&config, filename, snapshot_dir, *force);
+            }
+            Self::Prune {
+                snapshot_dir,
+                force,
+            } => {
+                prune(&config, snapshot_dir, *force);
+            }
+        }
+    }
+}
+
+fn list(config: &Config, snapshot_dir: &Option<PathBuf>) {
+    let snapshot_dir = snapshot_dir
+        .clone()
+        .unwrap_or_else(|| default_snapshot_dir(config));
+    println!("Snapshot dir: {:?}", snapshot_dir);
+    if let Ok(dir) = fs::read_dir(snapshot_dir) {
+        println!("\nLocal snapshots:");
+        for path in dir
+            .flatten()
+            .map(|entry| entry.path())
+            .filter(|p| p.is_file())
+        {
+            if let Some(Some(filename)) = path.file_name().map(|n| n.to_str()) {
```

Comment:
I believe you could omit one `Some` with [filter_map](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map). Most likely you'd be able to reach the print statement without any `if`s unless it hampers readability in your opinion.

Reply:
An owned `path` is needed below, I guess using `filter_map` will lose it?

Reply:
How about something like this?
```
        dir.flatten()
            .map(|entry| entry.path())
            .filter(|p| p.is_file() && p.ends_with(".car"))
            .for_each(|p| println!("{}", p.display()))
```

Reply:
Hmm I'm not a big fan of calling functions with side effects like `println` in lambdas but I guess `for_each` is for that purpose, also I just noticed there're 2 nested `Some` usage and one of them does not require an owned path, will fix as suggested

Reply:
Fixed with slight modification cuz [`PathBuf::ends_with`](https://doc.rust-lang.org/stable/std/path/struct.PathBuf.html#method.ends_with) does not work with extensions

```rust
use std::path::Path;

let path = Path::new("/etc/resolv.conf");

assert!(path.ends_with("resolv.conf"));
assert!(path.ends_with("etc/resolv.conf"));
assert!(path.ends_with("/etc/resolv.conf"));

assert!(!path.ends_with("/resolv.conf"));
assert!(!path.ends_with("conf")); // use .extension() instead
```

```rust
dir.flatten()
            .map(|entry| entry.path())
            .filter(|p| {
                p.is_file()
                    && p.extension()
                        .map(|ext| ext.to_str().unwrap_or_default() == "car")
                        .unwrap_or_default()
            })
            .for_each(|p| println!("{}", p.display()));
```